source/tutorials/zfs.rst

```diff
@@ -0,0 +1,299 @@
+.. _zfs:
+
+ZFS
+###
+
+This tutorial covers the setup of ZFS-on-Linux under Clear Linux,
+using a non-root device for your zpools. ZFS as a CLR_ROOT device
+is a not covered in this tutorial.
+
+.. contents::
+   :local:
+   :depth: 1
+
+Background
+**********
+
+Clear Linux does not ship with a binary ZFS kernel module (zfs.ko).
+ZFS is licensed under the CDDL. Linux is licensed under the GPL. These
+licenses are not compatible. Most Linux kernel developers assert that zfs.ko
+cannot be distributed in binary form alongside the kernel.
+
+Some distributions of Linux have a different opinion of
+CDDL / GPL compatibility -- most notably Canonical's Ubuntu -- and
+argue that a binary kernel module can be distributed alongside
+Linux without infringement. Clear Linux does not share that view.
+Therefore, zfs.ko must be built by the user under Clear Linux.
+
+It is worth understanding the `argument against shipping zfs.ko along
+with Linux`_ in addition to `the argument in favor`_.
+
+If you follow this tutorial and build zfs.ko on your system,
+*you should not redistribute that work in binary form*.
+
+Note that there is no issue with distributing ZFS in
+*source* form alongside Linux. According to those who argue
+both against and in favor of binary distribution, the CDDL and
+GPL are not incompatible with respect to redistribution of
+source code.
+
+There is nothing stopping a user from downloading the ZFS drivers,
+compiling them, then installing them for their own use. This is no
+different from downloading and using proprietary drivers like the Nvidia
+Graphics or Broadcom wireless drivers.
+
+Since the binaries for ZFS cannot be distributed with the
+Linux kernel binaries, you'll have some work to  ensure compatbility
+when new kernels are released.
+
+This is not always trivial, since new kernel release candidates will
+invariably break *something* in ZFS. The ZFS-on-Linux team is
+very good about catching up quickly, but on occasion you will find
+a lag between a new kernel release and a supported ZFS driver.
```

Comment:
We may want to recommend using an LTS kernel here, to reduce the likelihood of breaking changes between the kernel and ZFS.

Reply:
👍

Reply:
We may want to add that "users should test releases before deploying."

Reply:
I added some language that conveys these warnings.